### PR TITLE
Update get_dataset and make_evaluation_prediction

### PR DESCRIPTION
**Pull request by npnv**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/dataset/split/splitter.py [hunk 1]
```diff
@@ -43,6 +43,7 @@
 """
 
 from abc import ABC, abstractmethod
+from cProfile import label
```
Comment (lostella) [inline src/gluonts/dataset/split/splitter.py:46]: Not required? You can turn off automatic imports in the Python extension for VSCode

File: src/gluonts/evaluation/backtest.py [hunk 2]
```diff
@@ -30,14 +29,13 @@
...
-from gluonts.transform import AdhocTransform
 
 
 def make_evaluation_predictions(
-    dataset: Dataset,
+    dataset: TestIterable,
     predictor: Predictor,
     num_samples: int = 100,
-) -> Tuple[Iterator[Forecast], Iterator[pd.Series]]:
+    ) -> Tuple[Iterator[Forecast], Iterator[pd.Series]]:
```
Comment (lostella) [inline src/gluonts/evaluation/backtest.py:38]:
```suggestion
) -> Tuple[Iterator[Forecast], Iterator[pd.Series]]:
```

File: src/gluonts/evaluation/backtest.py [hunk 3]
```diff
@@ -63,48 +61,11 @@ def make_evaluation_predictions(
...
-
     return (
-        predictor.predict(dataset_trunc, num_samples=num_samples),
-        ts_iter(dataset),
-    )
-
+            predictor.predict(dataset.input, num_samples=num_samples),
+            dataset.label,
+        )
+
```
Comment (lostella) [inline src/gluonts/evaluation/backtest.py:68]: Indentation is strange here

**Review by lostella — CHANGES_REQUESTED** [review]

**Comment by lostella** [comment]
Hey @npnv, thanks for this! I suggested a few changes. In addition to that:
- some changes are required to `backtest_metrics`, for example type annotations, following the ones to `make_evaluation_predictions` probably
- some tests use `make_evaluation_prediction` or `backtest_metrics` with a `Dataset` object, which is not supported anymore with your changes, so those tests should be updated
- a thought: is `TestDataset` maybe a better name than `TestIterable`?
- the `TrainDatasets` class now doesn't have a `test` constructor argument anymore (it's a property); maybe we rename the constructor arguments to `train_dataset` and `test_dataset` and have a `train` and `test` property (the `test` one is there already)?
